auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,149 @@
+
+Both of these ways create a dependency on external software and might be complicated for beginners. Native support for this feature directly into Gz Sim would facilitate the effortless generation of accurate simulations.
+
+## User Perspective
+
+To specify the `<inertial>` element of a `<link>` in SDFormat, the user needs to add the `<mass>`, `<pose>`, and `<inertia>` tags. Here `<mass>` is the mass of the link and `<pose>` is the pose of the Centre of Mass with respect to the link frame. The `<inertia>` tag, on the other hand, needs to enclose the following 6 tags:
+
+ * `<ixx>` (Moment of Inertia)
+
+ * `<iyy>` (Moment of Inertia)
+
+ * `<izz>` (Moment of Inertia)
+
+ * `<ixy>` (Product of Inertia)
+ 
+ * `<ixz>` (Product of Inertia)
+ 
+ * `<izy>` (Product of Inertia)
+
+This proposal suggests the addition of the following in `SDFormat`: 
+
+ 1. An `auto` parameter for the `<inertia>` tag that would tell `libsdformat` to calculate Inertia matrix values automatically for the respective link. 
+
+ 2. A `//link/collision/material_density` tag for specifying the density of the collision geometry. This density value would be used to calculate the inertial parameters of the respective collision geometry. Adding this as part of the `<collision>` tag would allow a user to simulate links with different material types for different collisions. By default, the value of density would be set equal to that of water which is 1 kg/m^3.  
+
+The example snippet below shows how the above proposed elements would be used in a SDFormat `<link>`:
+
+```
+<link name="robot_link">
+  <inertial>
+    <inertia auto=”true” />
+  </inertial>
+  <collision name="collision">
+    <material_density>*some_float_value*</material_density>
+      <geometry>
+        .
+        .
+      </geometry>
+  </collision>
+  <visual name="visual">
+    <geometry>
+      .
+      .
+    </geometry>
+    <material>
+      .
+      .
+    </material>
+  </visual>
+</link>
+```
+
+## Proposed Implementation
+Below are some key architectural considerations for the implementation of this feature:
+
+ *  The parsing of the proposed SDFormat elements and the Moment of Inertia calculations for primitive geometries(Box, Cylinder, Sphere, Ellipsoid and Capsule) can be implemented as an integral part of `libsdformat`. This would help enable all simulators that rely on SDFormat to utilize this feature.
+
+ * In case of 3D meshes being used as geometries, a modular architecture can be followed where the user can integrate a custom Moments of Inertia calculator. Though for a default implementation of MOI calculations of 3D meshes, a [Voxelization-based](#inertia-matrix-calculation-with-voxelization-for-3d-mesh) method is proposed.
```

Comment:
Can you add a section on the API of the modular architecture for providing a MOI calculator function? Also, I believe even the voxelization method will not be implemented in libsdformat. It can be discussed here as a reference implementation, but I think discussing the API for the external MOI calculator is more important.

Reply:
Done in [fc2c6fb](https://github.com/gazebosim/sdf_tutorials/pull/92/commits/fc2c6fb8ab22e25cc3d265354eda3d2dab4a2f87)